var/ramble/repos/builtin/workflow_managers/slurm/workflow_manager.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import os
+
+from ramble.wmkit import *
+from ramble.expander import ExpanderError
+from ramble.application import experiment_status
+
+from spack.util.executable import Executable
+
+_ensure_job_id_snippet = r"""
+job_id=$(< {experiment_run_dir}/.slurm_job)
+if [ -z "${job_id:-}" ]; then
+    echo "No valid job_id found" 1>&2
+    exit 1
+fi
+"""
+
+
+class Slurm(WorkflowManagerBase):
+    """Slurm workflow manager"""
+
+    name = "slurm"
+
+    maintainers("linsword13")
+
+    tags("workflow", "slurm")
+
+    def __init__(self, file_path):
+        super().__init__(file_path)
+
+        self.runner = SlurmRunner()
+
+    workflow_manager_variable(
+        name="partition",
+        default="",
+        description="Name of the slurm partition for job submission",
+    )
+
+    workflow_manager_variable(
+        name="job_name",
+        default="{application_name}_{workload_name}_{experiment_name}",
+        description="Slurm job name",
+    )
+
+    workflow_manager_variable(
+        name="extra_sbatch_headers",
+        default="",
+        description="Extra sbatch headers added to slurm job script",
+    )
+
+    render_content(
+        name="batch_submit",
+        content_tpl=r"""
+#!/bin/bash
+sbatch {slurm_execute_experiment} | tee >(awk '{print $NF}' > {experiment_run_dir}/.slurm_job)
+        """.strip(),
+    )
+
+    render_content(name="query_job", content_func="_slurm_query_script")
+
+    def _slurm_query_script(self):
+        return rf"""
+#!/bin/bash
+{_ensure_job_id_snippet}
+{self.runner.generate_query_command("$job_id")}
+        """.strip()
+
+    render_content(name="cancel_job", content_func="_slurm_cancel_script")
+
+    def _slurm_cancel_script(self):
+        return rf"""
+#!/bin/bash
+{_ensure_job_id_snippet}
+{self.runner.generate_cancel_command("$job_id")}
+        """.strip()
+
+    render_content(
+        name="slurm_execute_experiment", content_func="_slurm_execute_script"
+    )
+
+    def _slurm_execute_script(self):
+        headers = ["#!/bin/bash"]
+        expander = self.app_inst.expander
+        # Adding pre-defined sets of headers
+        pragmas = [
+            ("#SBATCH -N {}", "n_nodes"),
+            ("#SBATCH -p {}", "partition"),
+            ("#SBATCH --ntasks-per-node {}", "processes_per_node"),
+            ("#SBATCH -J {}", "job_name"),
```

Comment:
Can you also add `--gpus-per-node` = `{gpus_per_node}` here?

Reply:
Definitely, added.